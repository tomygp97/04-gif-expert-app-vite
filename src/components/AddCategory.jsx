import { useState } from 'react';
import PropTypes from 'prop-types';  // Da error en vite

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState("");

  
  // const onImputChange = (e) => {
  //   setInputValue(e.target.value)
  // }  
  const onImputChange = ({target}) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault(); //Evita el refresh de la pag al enviar un submit

    if ( inputValue.trim().length <= 2 ) return;

    onNewCategory( inputValue.trim() );
    setInputValue("");  //Despues de enviar el submit borra en el form la cadena escrita y lo deja vacio para escribir nuevamente

  }

  return (
    <>
        <form onSubmit= {onSubmit} aria-label="form"> 

        <input 
            type= "text"
            placeholder= "Buscar Gifs"
            value= {inputValue}
            onChange= {onImputChange}
        />

        </form>

        {/* <button onClick= {handleAdd}>Add Valorant</button> */}

    </>
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}

export default AddCategory;
