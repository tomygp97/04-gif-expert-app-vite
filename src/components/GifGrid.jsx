// import { useEffect, useState } from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";
// import { getGifs } from "../helpers/getGifs";

export const GifGrid = ( {category} ) => {

    const { images, isLoading } = useFetchGifs( category );

 
    // const [images, setImages] = useState([]);

    // const getImages = async() => {
    //     const newImages = await getGifs( category );
    //     setImages(newImages);
    // }

    // useEffect( () => {
        
    //     getImages();
        
    // }, [] )


    return (
    <>
        <h3>{category}</h3>
        {
            isLoading && ( <h2>Cargando...</h2> )
        }


        <div className="card-grid">
            {
                images.map((image) => (
                   <GifItem key={image.id}
                   { ...image }
                   />
                ))
            }
 
            
        </div>

    </>
  )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}