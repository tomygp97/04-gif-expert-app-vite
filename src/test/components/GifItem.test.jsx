import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";



describe('Pruebas en <GifItem />', () => { 
   
    const title = "Saitama";
    const url = "https://one-punch.com/saitama.jpg";

    test('should match with snapshot', () => { 


       const {container} = render( <GifItem title={ title } url={ url } /> );
       expect (container).toMatchSnapshot();
     });

     
     test('should show the correct img with URL and ALT property', () => { 
         
         render( <GifItem title={ title } url={ url } /> );
         screen.debug();
        // expect( screen.getByRole("img").src).toBe( url );
        // expect( screen.getByRole("img").alt).toBe( title );

        const  { src, alt } = screen.getByRole("img");
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
         
     });

     test('should show title in the component', () => {

        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText(title) ).toBeTruthy();

     })
        
        
        
    });