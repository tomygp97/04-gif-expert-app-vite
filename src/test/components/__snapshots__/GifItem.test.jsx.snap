// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
